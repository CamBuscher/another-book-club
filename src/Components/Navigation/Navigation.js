import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import SignOutButton from '../Authentication/SignOut';

import * as routes from '../../constants/routes';

const Navigation = ({ authUser, history }) => 
  <div>
    {authUser
      ? <NavigationAuth history={history}/>
      : <NavigationNonAuth />
    }
  </div>

const NavigationAuth = ({ history }) =>
  <ul className='nav-bar'>
    <li><Link to={routes.ACCOUNT}>Your book clubs</Link></li>
    <li><Link to={routes.ALL_CLUBS}>Browse all clubs</Link></li>
    <li><SignOutButton history={history}/></li>
  </ul>

const NavigationNonAuth = () =>
  <ul className='nav-bar'>
    <li><Link to={routes.SIGN_IN}>Sign In</Link></li>
  </ul>

export default Navigation;