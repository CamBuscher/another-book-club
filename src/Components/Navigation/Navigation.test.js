import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';

describe('Navigation', () => {
  let wrapper;
   
  it('should match snapshot when given no authUser', () => {
    wrapper = shallow(<Navigation authUser={false} history={{}}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot when given an authUser', () => {
    wrapper = shallow(<Navigation authUser={true} history={{}} />);

    expect(wrapper).toMatchSnapshot();
  });
});