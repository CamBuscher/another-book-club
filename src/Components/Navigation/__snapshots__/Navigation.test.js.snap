// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot when given an authUser 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    authUser={true}
    history={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <NavigationAuth
        history={Object {}}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "history": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <NavigationAuth
          history={Object {}}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "history": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Navigation should match snapshot when given no authUser 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    authUser={false}
    history={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <NavigationNonAuth />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <NavigationNonAuth />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
